src/core/reactor.cc

```diff
@@ -4748,9 +4747,12 @@ std::chrono::nanoseconds reactor::total_steal_time() {
     // process is ready to run but the kernel hasn't scheduled us yet, that would be technically
     // steal time but we have no ways to account it.
     //
+    // Furthermore, not all steal is from other processes: time used by the syscall thread and any
+    // alien threads will show up as steal as well as any time spent in a system call that
+    // unexpectedly blocked (since CPU time won't tick up when that occurs).
+    //
     // But what we have here should be good enough and at least has a well defined meaning.
-    return std::chrono::duration_cast<std::chrono::nanoseconds>(now() - _start_time - _total_sleep) -
-           std::chrono::duration_cast<std::chrono::nanoseconds>(thread_cputime_clock::now().time_since_epoch());
+    return total_awake_time() - total_cpu_time();
```

Comment:
I'd really appreciate if this change went to previous (or separate, whichever is simpler for you) patch. It has nothing to do with updating the way _total_sleep is calculated, but is just re-using the helper methods introduced by previous change

Reply:
Will do.

Reply:
I have moved this to the prior commit in the most recent push.